persist/fs/mmap.go

```diff
@@ -81,7 +81,6 @@ func mmapFiles(opener fileOpener, files map[string]mmapFileDesc) (mmapFilesResul
 		}
 		if result.warning != nil {
 			multiWarn = multiWarn.Add(errorWithFilename(filePath, result.warning))
-			break
```

Comment:
Ha, interesting.  Maybe we should add a unit test for this?

We can use some package level indirection to mock this out.

e.g. 
```go
var mmapFileFn = mmapFile

// ...
func TestMmapFiles(t *testing.T) {
  currMmapFileFn := mmapFileFn
  mmapFileFn = func(..) {
    // mock..
  }
  defer func() {
    mmapFileFn = currMmapFileFn
  }()
}
```

Reply:
I try and avoid those if I can cause they break parallel tests. I'll add a unit test though

Reply:
We only run package tests in parallel, but sure, you can probably avoid it if you'd like with: